hook/hook.go

```diff
@@ -0,0 +1,50 @@
+package hook
+
+import (
+	"sync"
+
+	"github.com/flamego/flamego"
+	"github.com/hashicorp/go-multierror"
+	flag "github.com/spf13/pflag"
+)
+
+// Hooker is the interface for the webhook originator
+type Hooker interface {
+	prepare() error
+	register(fs *flag.FlagSet, f *flamego.Flame, path string)
+}
+
+var (
+	hooksMu sync.RWMutex
+	hooks   = make(map[string]Hooker)
```

Comment:
nit: My brain feels a bit split here
```suggestion
	hookersMu sync.RWMutex
	hookers   = make(map[string]Hooker)
```

WDYT?